src/expand-envs.ts

```diff
@@ -4,8 +4,8 @@
  * the environment variable doesn't exist, it leaves it as is.
 */
 export function expandEnvs (str: string, envs: { [key: string]: any }): string {
-  return str.replace(/(?<!\\)\$[a-zA-Z0-9_]+/g, varName => {
-    const varValue = envs[varName.slice(1)]
-    return varValue === undefined ? varName : varValue
+  return str.replace(/(?<!\\)\$\{?([a-zA-Z0-9_]+)\}?/g, (_, varName: string) => {
```

Comment:
The current implementation does not check, if the pattern is contained in a string. That would make the regex way more complex.

Currently something like `env-cmd -x ${WHO}` would also be expanded. I don’t think that is a big deal, but if so we could use something like `((?<="[^"]*)\${([a-zA-Z0-9_]+)}(?=[^"]*")|(?<!\\)\$([a-zA-Z0-9_]+))`.